dashboard/test/controllers/lti_provider_controller_test.rb

```diff
@@ -0,0 +1,206 @@
+require 'test_helper'
+
+class LtiProviderControllerTest < ActionController::TestCase
+  include Mocha::API
+
+  TEST_PROVIDER_KEY = "dc3872a4b605f1f36242a837172ce2c0"
+  TEST_CONSUMER_KEY = "f10ee9fc082219227976f2c1603a3d77"
+
+  setup do
+    CDO.stubs(:lti_credentials).returns({TEST_CONSUMER_KEY => TEST_PROVIDER_KEY})
+  end
+
+  def lti_consumer_params(consumer_key, consumer_secret, test_name, *args)
+    # match URL used by MiniTest post :sso call
+    tc = IMS::LTI::ToolConfig.new(title: 'Test LTI Provider',
+                                  launch_url: 'http://test.host/auth/lti')
+
+    @consumer = IMS::LTI::ToolConsumer.new(
+      consumer_key,
+      consumer_secret,
+      args[0])
+    @consumer.set_config(tc)
+
+    # Set some launch data, see:
+    # http://www.imsglobal.org/LTI/v1p1pd/ltiIMGv1p1pd.html#_Toc309649684
+    #
+    # Only this first attribute is required, the rest are recommended.
+    # Just use the MiniTest test name
+    @consumer.resource_link_id = test_name
+
+    @consumer.generate_launch_data
+  end
+
+  # For MiniTest need to have params passed through both query params and
+  # request body
+  def lti_post(action, params)
+    @request.headers['Content-Type'] = 'application/x-www-form-urlencoded'
+    post action, params.to_param, params
+  end
```

Comment:
nit: prefer test helper methods to be below the test cases and private. Not a big deal though. [example](https://github.com/code-dot-org/code-dot-org/blob/d82be354e9430f12b0d7296e582b75a2a3a829d7/dashboard/test/models/activity_test.rb#L101-L109)